extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -147,11 +151,17 @@ private String ParseFullFile(RDFParser[] parsers, InputStream[] in, String baseu
         return format;
     }
 
-    private String getFormatFromUrl(String url, InputStream[] i)
+    private String getFormatFromUrl(ArrayList<RDFParser> parsers, String url, InputStream[] i)
             throws IOException, MalformedURLException {
         StringBuffer response = new StringBuffer();
         URL obj = new URL(url);
         HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+        List<RDFFormat> rdfFormats = parsers.stream().map(val -> val.getRDFFormat()).collect(Collectors.toList());
```

Comment:
You can get the formats more easily by doing [`RDFParserRegistry.getInstance().getKeys()`](http://docs.rdf4j.org/javadoc/latest/org/eclipse/rdf4j/common/lang/service/ServiceRegistry.html#getKeys--).